emu/src/arm7tdmi.rs

```diff
@@ -221,7 +228,15 @@ impl Arm7tdmi {
             }
         };
 
-        for reg_source in 0..16 {
+        // If we are decreasing we still want to store the lowest reg to the lowest
+        // memory address. For this reason we reverse the loop order.
+        let mut regs_it: Box<dyn DoubleEndedIterator<Item = u8>> = Box::new(0..16);
+
+        if !up_down {
+            regs_it = Box::new(regs_it.rev());
+        }
```

Comment:
I think is more clear and easy something like
```rust
let range_registers = if up_down {
            0..=15
        } else {
            15..=0
        };
```

Reply:
ahahah definitely! I don't know why I went down that way

Reply:
Mh ok I've tried it but it looks like `15..=0` doesn't work (it's an empy range). One should do `(0..=15).rev()` but then it has a different type than `0..=15` and it doesn't work

Reply:
Ok then we can go for
```rust
    let x: Box<dyn Iterator<Item = usize> = if b > a {
        x = Box::new(0..=15)
    } else {
        x = Box::new((0..=15).rev())
    };
```